common/unsafe/src/main/java/org/apache/spark/sql/catalyst/util/CollationFactory.java

```diff
@@ -36,11 +36,57 @@
  * Provides functionality to the UTF8String object which respects defined collation settings.
  */
 public final class CollationFactory {
+
+  /**
+   * Identifier for single a collation.
+   */
+  public static class CollationIdentifier {
+    public final String provider;
+    public final String name;
+    public final String version;
+
+    public CollationIdentifier(String provider, String collationName, String version) {
+      this.provider = provider;
+      this.name = collationName;
+      this.version = version;
+    }
+
+    public static CollationIdentifier fromString(String identifier) {
+      long numDots = identifier.chars().filter(ch -> ch == '.').count();
+      assert(numDots > 0);
+
+      if (numDots == 1) {
+        String[] parts = identifier.split("\\.", 2);
+        return new CollationIdentifier(parts[0], parts[1], null);
+      }
+
+      String[] parts = identifier.split("\\.", 3);
+      return new CollationIdentifier(parts[0], parts[1], parts[2]);
+    }
+
+    @Override
+    public String toString() {
+      if (version != null) {
+        return String.format("%s.%s.%s", provider, name, version);
+      }
+
+      return toStringWithoutVersion();
+    }
+
+    /**
+     * Returns the identifier's string value without the version.
+     */
+    public String toStringWithoutVersion() {
```

Comment:
can we add more comments to explain when we should call `toStringWithoutVersion` instead of `toString`?

Reply:
added